### ISSUE TITLE
Support for Azure Machine Learning Compute Instance (new resource)

### ISSUE THREAD

**Issue by gro1m** [hashicorp/terraform-provider-azurerm#11255]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure offers Compute Instance to run ML trainings in an explorative setting with IDE capabilities.

A general description of the Azure Machine Learning Workflow can be found here: https://docs.microsoft.com/en-us/azure/machine-learning/how-to-create-attach-compute-studio. Would be cool if these resources were to provided via Terraform (but probably also a huge effort)

### New or Affected Resource(s)

* azurerm_machine_learning_compute_instance

### Potential Terraform Configuration

```hcl
resource "azurerm_machine_learning_compute_instance" "ml_ci" {
workspace_name = "aml-workspace"
compute_name = "my-compute-instance"
container_registry_name = "myacr"
key_vault_name = "mykv"
app_insights_name = "myappinsights"
location = "westeurope"
(sku = "Basic")
tags = {"department": "IT", "cost center" : "mybank"}
vnet_rg_name = "myrg"
vnet_name = "myvnet"
subnet_name = "mysubnet"
remoteLoginPortPublicAccess = "Disabled"
vm_size = "Standard_NC6" #gpu
vm_priority = "Dedicated"
access = [<object_ids of users or managed identites>] #maybe a set would be better, as order is not really necessary here
initial_run_status = "Running" #"Idle", ("Stopped" would probably not make sense)
run_configuration = {source_directory = <project folder>, script = "train.py", compute_target="my-compute", environment = ...}
}
```
It is of course disputable if the ability to even submit a run to a compute instance can or shall be covered by iac, if it is feasible and if this should be included in the same resource. This refers to the `initial_run_status` and the `run_configuration` parameter.

### References

* https://github.com/Azure/azure-quickstart-templates/tree/master/101-machine-learning-compute-create-computeinstance (ARM template for compute instances)
* https://docs.microsoft.com/en-us/azure/machine-learning/how-to-set-up-training-targets (Training target if viable)

**Comment by gro1m** [comment]
Would address: https://github.com/terraform-providers/terraform-provider-azurerm/issues/6827

**Comment by gro1m** [comment]
Has been fixed by #12834
